simp_le.py

```diff
@@ -1183,5 +1210,33 @@ def test_error_exit_codes(self, dummy_stderr):
                 assert False
 
 
+class IOPluginTests(TestCase):
+    """Integration tests for plugin interface."""
+
+    # this is unittest suite | pylint: disable=missing-docstring
+
+    def setUp(self):
+        self.root = tempfile.mkdtemp()
+        self.key_path = os.path.join(self.root, 'key.pem')
+        test_key = os.path.join('testdata', "rsa2048_key.pem")
```

Comment:
I've tried that, but that seems to require so many indirections (implementation details) that I decided to use this approach instead. I have a small preference to fixed files for reproducibility and faster key generation.

Also, the current `IOPlugin.Data` thing does not scale, what about setting defaults to None? Something like (out of my head):

```py
IOPlugin.Data.__defaults__ = (None, ) * 4
```

Then it should become something like `key_data = IOPlugin.Data(key=gen_pkey(1024))` (+ some indirections, I could not get this to work).